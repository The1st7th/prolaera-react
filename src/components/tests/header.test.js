import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { renderEmail } from 'react-html-email';
import renderer from 'react-test-renderer';
import buildHeader, { Header } from '../header';
import writeFile from '../../helpers/writeFileHelper';

Enzyme.configure({ adapter: new Adapter() });

describe('buildHeader image resizing tests', async () => {
  it('resizes image successfully NHHco Logo', async () => {
    const Header = await buildHeader('http://assets.prolaera.com/NHHCo-lg.png');
    const headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });

  it('resizes image successfully KRS logo', async () => {
    const Header = await buildHeader('http://assets.prolaera.com/KRS-lg.png');
    const headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });

  it('resizes image successfully a-lign logo', async () => {
    const Header = await buildHeader(
      'https://prolaera.s3-us-west-2.amazonaws.com/sponsors/def4c2bf-0485-4eda-9b2b-0e4055ef0dbe.logo.png'
    );
    const headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });

  it.skip('it writes an html file to test.html', async () => {
    const Header = await buildHeader('http://assets.prolaera.com/NHHCo-lg.png');
    const headerHtml = renderEmail(<Header text={'View My Compliance'} link={'https://www.google.com/'} />);
    const saved = await writeFile(headerHtml);
    expect(saved).toEqual(true);
  });
});

describe('Header component tests', async () => {
  it('creates and checks snapshot of Header html', async () => {
    let headerComponent = renderer.create(<Header />);
    let headerJson = headerComponent.toJSON();
    expect(headerJson).toMatchSnapshot();
  });

  it('checks default Header html', async () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.contains(<a href="https://app.prolaera.com">prolaera.com</a>)).toBe(true);
  });

  it('checks custom link text in Header html', async () => {
    const wrapper = shallow(<Header text={'View My Compliance'} />);
    expect(wrapper.contains(<a href="https://app.prolaera.com">View My Compliance</a>)).toBe(true);
  });

  it('checks custom link url in Header html', async () => {
    const wrapper = shallow(<Header text={'View My Compliance'} link={'https://www.google.com/'} />);
    expect(wrapper.contains(<a href="https://www.google.com/">View My Compliance</a>)).toBe(true);
  });
});
