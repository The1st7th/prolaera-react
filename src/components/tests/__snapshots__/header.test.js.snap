// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests creates and checks snapshot of Header html 1`] = `
<table
  align="center"
  border="0"
  cellPadding={0}
  cellSpacing={0}
  className="emailHeader"
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
  valign="top"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
      >
        <table
          align="center"
          cellSpacing={0}
          style={
            Object {
              "marginBottom": "20px",
              "marginTop": "25px",
              "maxWidth": "584px",
              "padding": "15px",
            }
          }
        >
          <tr
            align="center"
          >
            <td
              width="292px"
            >
              <img
                alt="logo"
                height={47}
                src="https://assets.prolaera.com/prolaeraLogo_fullText.png"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "outline": "none",
                    "textDecoration": "none",
                  }
                }
                width={200}
              />
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`buildHeader image resizing tests resizes image successfully KRS logo 1`] = `
<table
  align="center"
  border="0"
  cellPadding={0}
  cellSpacing={0}
  className="emailHeader"
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
  valign="top"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
      >
        <table
          align="center"
          cellSpacing={0}
          style={
            Object {
              "maxWidth": "584px",
              "padding": "15px",
            }
          }
        >
          <tr
            align="left"
          >
            <td
              width="292px"
            >
              <img
                alt="logo"
                height={47}
                src="http://assets.prolaera.com/KRS-lg.png"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "outline": "none",
                    "textDecoration": "none",
                  }
                }
                width={209.45470692717583}
              />
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
              width="252px"
            >
              <div>
                <div />
                 
              </div>
              <div>
                <div />
                 
              </div>
              <div>
                <a
                  href="https://app.prolaera.com"
                >
                  prolaera.com
                </a>
                 
              </div>
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`buildHeader image resizing tests resizes image successfully NHHco Logo 1`] = `
<table
  align="center"
  border="0"
  cellPadding={0}
  cellSpacing={0}
  className="emailHeader"
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
  valign="top"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
      >
        <table
          align="center"
          cellSpacing={0}
          style={
            Object {
              "maxWidth": "584px",
              "padding": "15px",
            }
          }
        >
          <tr
            align="left"
          >
            <td
              width="292px"
            >
              <img
                alt="logo"
                height={47}
                src="http://assets.prolaera.com/NHHCo-lg.png"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "outline": "none",
                    "textDecoration": "none",
                  }
                }
                width={113.92391304347827}
              />
            </td>
            <td
              style={
                Object {
                  "textAlign": "center",
                }
              }
              width="252px"
            >
              <div>
                <div />
                 
              </div>
              <div>
                <div />
                 
              </div>
              <div>
                <a
                  href="https://app.prolaera.com"
                >
                  prolaera.com
                </a>
                 
              </div>
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`buildHeader image resizing tests resizes image successfully a-lign logo 1`] = `
<table
  align="center"
  border="0"
  cellPadding={0}
  cellSpacing={0}
  className="emailHeader"
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
  valign="top"
  width="100%"
>
  <tbody>
    <tr>
      <td
        align="center"
      >
        <table
          align="center"
          cellSpacing={0}
          style={
            Object {
              "marginBottom": "20px",
              "marginTop": "25px",
              "maxWidth": "584px",
              "padding": "15px",
            }
          }
        >
          <tr
            align="center"
          >
            <td
              width="292px"
            >
              <img
                alt="logo"
                height={47}
                src="https://prolaera.s3-us-west-2.amazonaws.com/sponsors/def4c2bf-0485-4eda-9b2b-0e4055ef0dbe.logo.png"
                style={
                  Object {
                    "border": "none",
                    "display": "block",
                    "outline": "none",
                    "textDecoration": "none",
                  }
                }
                width={199.08130081300814}
              />
            </td>
          </tr>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
